File: 2961/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(getGoodIndices([][]int{
		{2,3,3,10},
		{3,3,3,1},
		{6,1,1,4},
	}, 2))
	fmt.Println(getGoodIndices([][]int{
		{45,34,41,32},
		{9,2,15,3},
		{31,12,21,24},
	}, 1))
	fmt.Println(getGoodIndices([][]int{
		{39,3,1000,1000},
	}, 17))
}

func getGoodIndices(veriables [][]int, target int) []int {
	ans := []int{}
	for i, v := range veriables {

		abp := math.Pow(float64(v[0]), float64(v[1]))
		ab := math.Mod(abp, float64(10))

		abc := math.Pow(float64(ab), float64(v[2]))
		abcd := math.Mod(abc, float64(v[3]))

		if float64(abcd) == float64(target) {
			ans = append(ans, i)
		}
	}

	// return ans

	// ans := []int{}
	for i, v := range veriables {
		if pow_mod(pow_mod(v[0], v[1], 10), v[2], v[3]) == target {
			ans = append(ans, i)
		} 
	}
	return ans
}

func pow_mod(x, y, mod int) int {
	res := 1 
	for y > 0 {
		if (y & 1) == 1 {
			res = res * x % mod 
		}
		x = x * x % mod
		y >>= 1
	}
	return res
}

File: 2961/main.rs
struct Solution{}

impl Solution {
    pub fn get_good_indices(veriables: Vec<Vec<i32>>, target: i32) -> Vec<i32> {
        let mut ans: Vec<i32> = Vec::new();
        for i in 0..veriables.len() {
            if Self::pow_mod(Self::pow_mod(veriables[i][0], veriables[i][1], 10), veriables[i][2], veriables[i][3]) == target {
                ans.push(i as i32);
            }
        }
        ans
    }

    fn pow_mod(mut x: i32, mut y: i32, m: i32) -> i32 {
        let mut res = 1;
        while y > 0 {
            if (y & 1 ) == 1 {
                res = res * x % m;
            }
            x = x * x % m;
            y >>= 1;
        }
        res
    }
}

fn main() {
    println!("{:?}", Solution::get_good_indices(vec![vec![39,3,1000,1000]], 17))
}